library/vector.lisp

```diff
@@ -102,10 +103,10 @@
     (lisp (Vector :a) (v)
       (alexandria:copy-array v)))
 
-  (declare set-capacity (UFix -> Vector :a -> Unit))
-  (define (set-capacity new-capacity v)
+  (declare set-capacity! (UFix -> Vector :a -> Unit))
+  (define (set-capacity! new-capacity v)
     "Set the capacity of V to NEW-CAPACITY. Setting the capacity to lower then the length will remove elements from the end."
-    (let shrinking = (< new-capacity (length v)))
+    (let shrinking = (> new-capacity (length v)))
```

Comment:
I would write this as:

```
    (let shrinking = (< new-capacity (length v)))
    (lisp Unit (v shrinking new-capacity)
      ;; If the array is getting larger then dont change the
      ;; fill pointer
      (cl:adjust-array v new-capacity :fill-pointer shrinking)
```

Right now, the variable `shrinking` doesn't match the behavior (shrinking is True when it's actually expanding), though the code that follows it is correct.